server/api/blocks.go

```diff
@@ -463,31 +450,23 @@ func (a *API) handleUndeleteBlock(w http.ResponseWriter, r *http.Request) {
 
 	board, err := a.app.GetBoard(boardID)
 	if err != nil {
-		a.errorResponse(w, r.URL.Path, http.StatusInternalServerError, "", err)
-		return
-	}
-	if board == nil {
-		a.errorResponse(w, r.URL.Path, http.StatusNotFound, "", nil)
+		a.errorResponse(w, r, err)
 		return
 	}
 
 	block, err := a.app.GetLastBlockHistoryEntry(blockID)
 	if err != nil {
-		a.errorResponse(w, r.URL.Path, http.StatusInternalServerError, "", err)
-		return
-	}
-	if block == nil {
-		a.errorResponse(w, r.URL.Path, http.StatusNotFound, "", nil)
+		a.errorResponse(w, r, err)
 		return
 	}
 
 	if board.ID != block.BoardID {
-		a.errorResponse(w, r.URL.Path, http.StatusNotFound, "", nil)
+		a.errorResponse(w, r, model.NewErrNotFound("block"))
```

Comment:
Would prefer to have block id passed to ErrNotFound.

Reply:
Agree, not found errors have been addressed on https://github.com/mattermost/focalboard/pull/3792/commits/db85b38428363ada96685dcfa414306b2c43821d, and the other error types that were created without context on https://github.com/mattermost/focalboard/pull/3792/commits/8e27b1af33059b45fd9f6b295bcfee5347981b7c

Merged and adapted the new Card API endpoints on https://github.com/mattermost/focalboard/pull/3792/commits/8bb7d209e90d0bf32b81b57668d1535dc250ef86

Reply:
Thanks for updating the card APIs!